### PR TITLE
Refactoring syndication rights inference logic

### PR DESCRIPTION
**Pull request by marialivia16**
Following the issues we've seen with the inference of syndication rights in albums, I took some time to rethink the logic (see image below). Previously every update of an image was a message sent to thrall, which caused a lot of inconsistencies, resulting in some images in albums missing their inferred rights. 

syndcation_rights_logic

### CODE REVIEW

**Comment by mbarton** [comment]
The refactoring looks great, much much clearer.

As discussed IRL lets throw it on TEST and see how it does, especially with removing images from a shoot and the resulting loss of rights inference on the other images in the shoot.

We should also look to see if we can remove the `setRefresh(true)` calls on the queries, especially for [applyImageMetadataOverride](https://github.com/guardian/grid/blob/73e1d6c08c3b9301c83d55264f6306d8ea3b46d1/thrall/app/lib/ElasticSearch.scala#L249) since that will be called for things other than syndication. Since ElasticSearch is eventually-consistent we should avoid to much dependence on `setRefresh` that tries to pretend otherwise.

File: common-lib/src/main/scala/com/gu/mediaservice/model/Image.scala [hunk 1]
```diff
@@ -30,6 +30,8 @@ case class Image(
 ) {
   def rcsPublishDate: Option[DateTime] = syndicationRights.flatMap(_.published)
 
+  def isInferred: Boolean = syndicationRights.forall(_.isInferred == true)
```
Comment (mbarton) [inline common-lib/src/main/scala/com/gu/mediaservice/model/Image.scala]: Nitpicking but `isInferred` might be a confusing name on the whole `Image`. Maybe `hasInferredRights`?

File: thrall/test/syndication/SyndicationRightsOpsTest.scala [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    "Images in a photoshoot" - {
+      "correctly infer rights in the photoshoot when an image with syndication rights is added" in {
+        val photoshootTitle = Photoshoot(s"photoshoot-${UUID.randomUUID()}")
+
+        withPhotoshoot(photoshootTitle) { images =>
+          withImage(imageWithSyndRights) { imageWithRights =>
+            whenReady(syndRightsOps.upsertOrRefreshRights(image = imageWithRights, previousPhotoshootOpt = None, currentPhotoshootOpt = Some(photoshootTitle))) { _ =>
+              Thread.sleep(5000)
```
Comment (mbarton) [inline thrall/test/syndication/SyndicationRightsOpsTest.scala]: As discussed IRL, lets try to remove the sleeps in the test

File: common-lib/src/main/scala/com/gu/mediaservice/model/Image.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (akash1810) [reply common-lib/src/main/scala/com/gu/mediaservice/model/Image.scala]: the comment came back!

**Pull request merged**